packages/medusa/src/api/routes/admin/products/create-variant.ts

```diff
@@ -105,58 +119,126 @@ import { defaultFields, defaultRelations } from "./"
 export default async (req, res) => {
   const { id } = req.params
 
-  const schema = Validator.object().keys({
-    title: Validator.string().required(),
-    sku: Validator.string().allow(""),
-    ean: Validator.string().allow(""),
-    upc: Validator.string().allow(""),
-    barcode: Validator.string().allow(""),
-    hs_code: Validator.string().allow(""),
-    inventory_quantity: Validator.number().default(0),
-    allow_backorder: Validator.boolean().optional(),
-    manage_inventory: Validator.boolean().optional(),
-    weight: Validator.number().allow(null).optional(),
-    length: Validator.number().allow(null).optional(),
-    height: Validator.number().allow(null).optional(),
-    width: Validator.number().allow(null).optional(),
-    origin_country: Validator.string().allow(""),
-    mid_code: Validator.string().allow(""),
-    material: Validator.string().allow(""),
-    metadata: Validator.object().optional(),
-    prices: Validator.array()
-      .items(
-        Validator.object()
-          .keys({
-            region_id: Validator.string().empty(null),
-            currency_code: Validator.string().required(),
-            amount: Validator.number().integer().required(),
-            sale_amount: Validator.number().optional(),
-          })
-          .xor("region_id", "currency_code")
-      )
-      .required(),
-    options: Validator.array()
-      .items({
-        option_id: Validator.string().required(),
-        value: Validator.string().required(),
-      })
-      .default([]),
-  })
-
-  const { value, error } = schema.validate(req.body)
-  if (error) {
-    throw new MedusaError(MedusaError.Types.INVALID_DATA, error.details)
-  }
+  const validated = await validator(
+    AdminPostProductsProductVariantsReq,
+    req.body
+  )
 
-  const productVariantService = req.scope.resolve("productVariantService")
-  const productService = req.scope.resolve("productService")
+  const productVariantService: ProductVariantService = req.scope.resolve(
+    "productVariantService"
+  )
+  const productService: ProductService = req.scope.resolve("productService")
 
-  await productVariantService.create(id, value)
+  await productVariantService.create(id, validated)
 
   const product = await productService.retrieve(id, {
-    select: defaultFields,
-    relations: defaultRelations,
+    select: defaultAdminProductFields,
+    relations: defaultAdminProductRelations,
   })
 
   res.json({ product })
 }
+
+class ProductVariantOptionReq {
+  @IsString()
+  value: string
+
+  @IsString()
+  option_id: string
+}
+
+class ProductVariantPricesReq {
+  @Validate(XorConstraint, ["currency_code"])
+  region_id?: string
+
+  @Validate(XorConstraint, ["region_id"])
+  currency_code?: string
+
+  @IsNumber()
+  @Type(() => Number)
+  amount: number
```

Comment:
Nit: Since this is given as an integer in the OAS description, shouldn't we also use `@IsInt()` here? The same goes for the other uses of `@IsNumber()` in this file.